app/main.py

```diff
@@ -33,8 +34,21 @@ def image(file: UploadFile) -> dict[str, str]:
             status_code=BAD_REQUEST,
             detail="Invalid file content type for image.",
         )
-
     image = Image.open(file.file)
     text = convert_image_to_text(image)
 
     return {"text": text}
+
+
+@app.post("/docx")
+def docx(file: UploadFile) -> dict[str, str]:
+    """Accept a .docx file name and output the text within it."""
+    text = convert_docx_to_plain_text(file.filename)
+    return {"text": text}
+
+
+@app.post("/pdf")
+def pdf(file: UploadFile) -> dict[str, str]:
+    """Accept a PDF file name and output the text within it."""
+    pdf = convert_pdf_to_text(file.filename)

```

Comment:
Minor: could you make the naming between `convert_docx_to_plain_text` and `convert_pdf_to_text` consistent? Otherwise it might seem that docx is "plain text" while pdf is something else.

Reply:
Also minor: could you name the variable on this line `text` instead of `pdf`? `pdf` sounds like it would represent the entire document in memory.

Reply:
Yeah, sounds good.